simulator/mips/mips_instr.h

```diff
@@ -178,11 +178,11 @@ class FuncInstr
         bool complete = false;
 
         /* info for branch misprediction unit */
-        const bool predicted_taken = false;     // Predicted direction
-        const Addr predicted_target = NO_VAL32; // PC, predicted by BPU
+        bool predicted_taken = false;     // Predicted direction
+        Addr predicted_target = NO_VAL32; // PC, predicted by BPU
         bool _is_jump_taken = false;      // actual result
 
-        const Addr PC = NO_VAL32;
+        Addr PC = NO_VAL32;
```

Comment:
I guess we can return `const`s back, cannot we?

Reply:
I have removed const because a copy assignment is needed in std::unordered_map.

Reply:
It is needed only for `insert_or_assign` method: https://godbolt.org/g/2dUvN2
Actually we never assign the value in InstrCache, we either add a new key or invalidate the existing one and insert a value.